src/server.cob

```diff
@@ -388,16 +400,29 @@ ServerLoop.
             STOP RUN RETURNING 1
         END-IF
 
+        *> Send debug sample for the past tick
+        PERFORM SendDebugSamples
+
+        CALL "SystemTimeMicros" USING CURRENT-TIME
+        COMPUTE TICK-MAIN-DURATION = CURRENT-TIME - TICK-STARTTIME
+
         *> Read console command
         PERFORM ConsoleInput
 
         *> The remaining time of this tick can be used for accepting connections and receiving packets.
         CALL "SystemTimeMicros" USING CURRENT-TIME
+        MOVE 0 TO TICK-SLEEP-DURATION
         PERFORM UNTIL CURRENT-TIME >= TICK-ENDTIME
             PERFORM NetworkRead
             CALL "SystemTimeMicros" USING CURRENT-TIME
         END-PERFORM
 
+        *> debug samples are measured in nanoseconds
+        COMPUTE DEBUG-SAMPLE-FULL = (CURRENT-TIME - TICK-STARTTIME) * NANOSECOND-SCALE
+        COMPUTE DEBUG-SAMPLE-MAIN = TICK-MAIN-DURATION * NANOSECOND-SCALE
+        COMPUTE DEBUG-SAMPLE-IDLE = TICK-SLEEP-DURATION * NANOSECOND-SCALE
+        COMPUTE DEBUG-SAMPLE-TASKS = DEBUG-SAMPLE-FULL - DEBUG-SAMPLE-MAIN - DEBUG-SAMPLE-IDLE
```

Comment:
I'd suggest to put this into a single `COMPUTE` (maybe one line for each element, if you like ending with an inline comment) and in any case to only do the multiplication at the end.

Reply:
Thanks for taking a look.

What would be the advantage of doing the multiplication at the end? There is no way around doing it three times, as all four values are sent to the client.

Reply:
Ah, I haven't spot that _all_ are sent, in this case the current logic is of course the right one.
Sorry for the noise.